packages/rum-recorder/src/boot/rumRecorderPublicApi.ts

```diff
@@ -72,7 +67,33 @@ export function makeRumRecorderPublicApi(startRumImpl: StartRum, startRecordingI
       lifeCycle.notify(LifeCycleEventType.RECORD_STOPPED)
     }
 
+    onInit(userConfiguration)
+
     return startRumResult
   })
-  return rumRecorderGlobal
+
+  let onInit = (userConfiguration: RumRecorderUserConfiguration) => {
+    if (!userConfiguration.manualSessionReplayRecordingStart) {
+      startSessionReplayRecordingImpl()
+    }
+  }
+
+  let startSessionReplayRecordingImpl = () => {
+    onInit = () => startSessionReplayRecordingImpl()
+  }
+
+  let stopSessionReplayRecordingImpl = () => {
+    onInit = noop
+  }
+
+  const rumRecorderPublicApi = {
```

Comment:
Should we name this one rumRecorderGlobal to be consistent with rumGlobal on the RUM side?

Reply:
IMO, we should not. How the returned value will be used is up to the function caller. It will be used as a global and a exported value by `recorder.entry`, but I don't think the name of this internal variable should reflect that. I think it would make sense only if the function was called `makeRumRecorderGlobal`

Reply:
So it is maybe `rumGlobal` in `makeRumPublicApi` that we should rename `rumPublicApi` then, wdyt?

Reply:
Agreed, I renamed a few variables to `rumPublicApi` / `rumRecorderPublicApi` for consistency